packages/cli/src/lib/snapshot/snapshotReporter.ts

```diff
@@ -3,15 +3,50 @@ import {
   ResourceSnapshotsReportOperationModel,
   ResourceSnapshotsReportResultCode,
   ResourceSnapshotsReportStatus,
+  ResourceSnapshotType,
 } from '@coveord/platform-client';
 import {
   ReportViewerOperationName,
   ReportViewerResourceReportModel,
+  SnapshotReportStatus,
 } from './reportPreviewer/reportPreviewerDataModels';
 
 type ResourceEntries = [string, ResourceSnapshotsReportOperationModel];
 
+type NoopHandler = (this: void) => void;
+type SnapshotReporterHandler = (this: SnapshotReporter) => void | Promise<void>;
+
+type SnapshotReporterHandlers = FixableErrorHandlers &
+  UnfixableErrorHandlers &
+  SuccessfulReportHandler;
+
+type FixableErrorHandlers = {
+  [SnapshotReportStatus.MISSING_VAULT_ENTRIES]: /**
+   * TODO CDX-936: Define return type of the cb, & remove the SnapshotReporterHandler.
+   * Return type should be either a boolean if we do a 'all or nothing'
+   * handling (i.e. either we manage to do them all, or we do nothing)
+   * Return type should be a `Set<string>` if we do a 'best effort' strategy
+   * (i.e. if some entries are messed up, we ignore them and push the others)
+   */
+  ((this: SnapshotReporter) => boolean) | SnapshotReporterHandler | NoopHandler;
+};
+
+type UnfixableErrorHandlers = {
+  [K in SnapshotReportStatus]: SnapshotReporterHandler | NoopHandler;
+};
+
+type SuccessfulReportHandler = {
+  [K in SnapshotReportStatus]: SnapshotReporterHandler | NoopHandler;
+};
```

Comment:
May look daunting, but it helps when you call `setReportHandler`: You'll know what your callback can use.